components/organisms/InsightPage/InsightPage.tsx

```diff
@@ -473,18 +475,30 @@ const InsightPage = ({ edit, insight, pageRepos }: InsightPageProps) => {
             onSearch={(search) => setRepoSearchTerm(search as string)}
           />
 
-          <div>
+          <div className="w-full flex gap-3 md:items-center flex-col md:flex-row">
             <Button
               disabled={
-                repos.length === insightRepoLimit ||
+                repos.length >= insightRepoLimit! ||
                 (addRepoLoading.repoName === repoSearchTerm && addRepoLoading.isLoading)
               }
               loading={addRepoLoading.repoName === repoSearchTerm && addRepoLoading.isLoading}
               onClick={handleAddRepository}
               variant="outline"
+              className="shrink-0 w-max"
             >
               Add Repository
             </Button>
+
+            {repos.length >= insightRepoLimit! && insightRepoLimit! < 50 ? (
+              <span className="text-sm">
+                Your insight pages are limited to
+                <strong className="text-sauced-orange"> {insightRepoLimit}</strong> repos,{" "}
+                <Link href={`/user/settings#upgrade`} className="underline text-sauced-orange">
+                  upgrade
+                </Link>{" "}
+                to increase limit
+              </span>
+            ) : null}
```

Comment:
refactor (blocking): Add a [role=alert](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alert_role) so people using assistive technologies will be notified when they hit the insight page repo limit.

nit (wording): instead of `increase limit`, use `increase the limit`.

```suggestion
            <span role="alert" className="text-sm">
              {repos.length >= insightRepoLimit! && insightRepoLimit! < 50 ? (
                <p>
                  Your insight pages are limited to
                  <strong className="text-sauced-orange"> {insightRepoLimit}</strong> repos,{" "}
                  <Link href={`/user/settings#upgrade`} className="underline text-sauced-orange">
                    upgrade
                  </Link>{" "}
                  to increase the limit
                </p>
              ) : null}
            </span>
          </div>
```